samples/csharp/getting-started/Ranking_PersonalizedSort/README.md

```diff
@@ -0,0 +1,215 @@
+
+ Refer to the following code which shows how to split the data:
+
+ ```CSharp
+// When splitting the data, 20% is held for the test dataset.
+// To avoid label leakage, the GroupId (e.g. search\query id) is specified as the samplingKeyColumnName.  
+// This ensures that if two or more hotel instances share the same GroupId, that they are guaranteed to appear in the same subset of data (train or test).
+TrainTestData trainTestData = mlContext.Data.TrainTestSplit(labeledData, testFraction: .2, samplingKeyColumnName: nameof(HotelData.GroupId), seed: 1);
+IDataView trainData = trainTestData.TrainSet;
+IDataView testData = trainTestData.TestSet;
+
+// Save the test dataset to a file to make it faster to load in subsequent runs.
+using (var fileStream = File.Create(trainDatasetPath))
+{
+    mlContext.Data.SaveAsText(trainData, fileStream, separatorChar: ',', headerRow: true, schema: true);
+}
+
+// Save the train dataset to a file to make it faster to load in subsequent runs.
+using (var fileStream = File.Create(testDatasetPath))
+{
+    mlContext.Data.SaveAsText(testData, fileStream, separatorChar: ',', headerRow: true, schema: true);
+}
+`````
+
+### 3. Train Model
+This sample trains the model using the LightGbmRankingTrainer which relies on the LightGbm Lambdarank algorithm.  The model requires the following input columns:
+
+* Group Id - Column that contains the group id for each data instance.  Data instances are contained in logical groupings and each group has an identifier known as the group id.  In the case of the Expedia dataset, hotel search results are grouped by their corresponding query where the group id corresponds to the query or search id.  The input group id data type must be [key type](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.data.keydataviewtype). 
+* Label - Column that contains the deal rank (e.g. degree of relevance) of each data instance where higher values indicate higher relevance.  The input label data type must be [key type](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.data.keydataviewtype) or [Single](https://docs.microsoft.com/en-us/dotnet/api/system.single). 
+* Features - The columns that are influential in determining the relevance\rank of a data instance.  The input feature data must be a fixed size vector of type [Single](https://docs.microsoft.com/en-us/dotnet/api/system.single).
+
+When the trainer is set, **custom gains** are used to apply weights to each of the labeled rank values.  As described earlier in the sample, the potential label rank values are {0, 1, 2} which directly correlates to the specified gains {0, 1, 5}.  This helps to ensure that the model places more emphasis on ranking hotel search results labeled with 2 (e.g. signifies the user purchased\booked the hotel) so that they are positioned higher when compared to results labeled with 0 or 1.    
```

Comment:
Extra spaces:
```suggestion
When the trainer is set, **custom gains** are used to apply weights to each of the labeled rank values. As described earlier in the sample, the potential label rank values are {0, 1, 2} which directly correlates to the specified gains {0, 1, 5}. This helps to ensure that the model places more emphasis on ranking hotel search results labeled with 2 (e.g. signifies the user purchased\booked the hotel) so that they are positioned higher when compared to results labeled with 0 or 1.
```